File: src/SqlStudio.Core/Models/ConnectionInfo.cs
namespace SqlStudio.Core.Models;

public class ConnectionInfo
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public string Name { get; init; } = string.Empty;
    public string Server { get; init; } = string.Empty;
    public string Database { get; init; } = "master";
    public ConnectionAuthType AuthType { get; init; }
    public string? Username { get; init; }
    public string? Password { get; init; }
    public string? TenantId { get; init; }
    public bool TrustServerCertificate { get; init; }
    public int ConnectionTimeoutSeconds { get; init; } = 30;
    public int CommandTimeoutSeconds { get; init; } = 60;
    public bool Encrypt { get; init; } = true;
    public int Port { get; init; } = 1433;
}

File: SQLExplorer/Models/ConnectionInfo.swift
import Foundation

enum ConnectionAuthType: String, Codable, CaseIterable, Identifiable {
    case sqlAuthentication = "SQL Authentication"
    case entraIdInteractive = "Entra ID Interactive"

    var id: String { rawValue }
}

struct ConnectionInfo: Identifiable, Codable {
    var id: UUID = UUID()
    var name: String = ""
    var server: String = ""
    var port: Int = 1433
    var database: String = "master"
    var authType: ConnectionAuthType = .entraIdInteractive
    var username: String?
    var password: String?
    var tenantId: String?
    var trustServerCertificate: Bool = true
    var encrypt: Bool = true
}
